src/Cli/AzureCliLogin.ts

```diff
@@ -95,11 +88,86 @@ export class AzureCliLogin {
         console.log(`Done registering cloud: "${this.loginConfig.environment}"`)
     }
 
+    async loginWithSecret() {
+        if (this.isSuccess || !(this.loginConfig.servicePrincipalId && this.loginConfig.tenantId && this.loginConfig.servicePrincipalKey)) {
+            core.debug('Skip login with secret.');
+            return;
+        }

```

Comment:
It is not a programming convention to put the functionality for routing into each execution function `loginWith...`. Execution functions should be definitive and mutually exclusive.

We may take Azure CLI's source code as references:

https://github.com/Azure/azure-cli/blob/d331edf76e5866cda1e6e9641d935256fc4f03ff/src/azure-cli/azure/cli/command_modules/profile/custom.py#L119-L122

https://github.com/Azure/azure-cli/blob/060b414d8ac5e4d8f4fc17cc2e23c2154c762226/src/azure-cli-core/azure/cli/core/_profile.py#L146-L159